### DISCUSSION TITLE
Use pyg graph neural network for text classification

### DISCUSSION THREAD

**Discussion by yaoysyao** [pyg-team/pytorch_geometric#7133] [Q&A]
My current research field is text classification, and now I want to use PYG to build a graph neural network for text classification, which has the following two problems: 
1. Does PYG support text classification tasks? 
2. I want to use my own text dataset for research, is there any relevant example of how to use my own text dataset to construct code that conforms to the PYG data format, I looked at the demo in the source code and did not seem to find a task related to text classification.

**Comment by rusty1s** [comment]
In the end, PyG is a graph learning library. As long as you can model your text classification as a graph, you should be able to leverage PyG/GNNs to obtain node, link or graph embeddings. It is also possible to combine PyG with NLP models, e.g., models from the HuggingFace model hub.
